samcli/lib/sync/infra_sync_executor.py

```diff
@@ -0,0 +1,404 @@
+
+        Returns
+        -------
+        List[str]
+            The list of resource IDs that got changed during sanitization
+        """
+
+        resources = template_dict.get("Resources", {})
+        processed_resources: Set[str] = set()
+
+        for resource_logical_id in resources:
+            resource_dict = resources.get(resource_logical_id, {})
+
+            # Built resource dict helps with determining if a field is a local path
+            if built_template_dict:
+                built_resource_dict = built_template_dict.get("Resources", {}).get(resource_logical_id, {})
+
+            resource_type = resource_dict.get("Type")
+
+            if resource_type in CODE_SYNCABLE_RESOURCES or resource_type in SYNCABLE_STACK_RESOURCES:
+                processed_resource = self._remove_resource_field(
+                    resource_logical_id,
+                    resource_type,
+                    resource_dict,
+                    linked_resources,
+                    built_resource_dict if built_template_dict else None,
+                )
+
+                if processed_resource:
+                    LOG.debug("Sanitized %s resource %s", resource_type, resource_logical_id)
+                    processed_resources.add(processed_resource)
+
+            # Remove SamResourceId metadata since this metadata does not affect any cloud behaviour
+            resource_dict.get("Metadata", {}).pop("SamResourceId", None)
+            if not resource_dict.get("Metadata"):
+                resource_dict.pop("Metadata", None)
+            LOG.debug("Sanitizing the Metadata for resource %s", resource_logical_id)
+
+        return cast(Set, sorted(processed_resources))
+
+    def _remove_resource_field(
+        self,
+        resource_logical_id: str,
+        resource_type: str,
+        resource_dict: Dict,
+        linked_resources: Set[str] = set(),
+        built_resource_dict: Optional[Dict] = None,
+    ) -> Optional[str]:
+        """
+        Helper method to process resource dict
+
+        Parameters
+        ----------
+        resource_logical_id: str
+            Logical ID of the resource
+        resource_type: str
+            Resource type
+        resource_dict: Dict
+            The resource level dict containing Properties field
+        linked_resources: Set[str]
+            The corresponding resources in the other template that got processed
+        built_resource_dict: Optional[Dict]
+            Only passed in for current template sanitization to determine if local
+
+        Returns
+        -------
+        Optional[str]
+            The processed resource ID
+        """
+        processed_logical_id = None
+
+        if resource_type == AWS_LAMBDA_FUNCTION:
+            for field in LAMBDA_FUNCTION_REMOVAL_MAP.get(resource_type, {}).get("Code", []):
+                # We sanitize only if the provided resource is local
+                if (
+                    built_resource_dict
```

Comment:
Why we are having `built_resource_dict` in the `if` sentences, given that it can't be `None` (its type in the method definition is not `Optional`). Can we remove that to simplify the statement?

Reply:
It can be none, 
```
                processed_resource = self._remove_resource_field(
                    resource_logical_id,
                    resource_type,
                    resource_dict,
                    linked_resources,
                    built_resource_dict if built_template_dict else None,
                )
```
When sanitizing for the deployed template we do not pass in the build resource dict

Reply:
built_resource_dict: Optional[Dict] = None, this is what's in the method definition